### PR TITLE
Make Lisp_Object a proper struct type

### PR DESCRIPTION
**Pull request by shaleh**

### CODE REVIEW

File: rust_src/src/lisp.rs [hunk 1]
```diff
@@ -95,12 +95,12 @@ impl LispObject {
...
 
     #[inline]
-    pub fn from_raw(i: EmacsInt) -> LispObject {
+    pub fn from_raw(i: Lisp_Object) -> LispObject {
         LispObject(i)
     }
 
     #[inline]
-    pub fn to_raw(self) -> EmacsInt {
+    pub fn to_raw(self) -> Lisp_Object {
```
Comment (db48x) [inline rust_src/src/lisp.rs]: I should have done that myself, just to make it clearer.

File: rust_src/remacs-sys/lib.rs [hunk 2]
```diff
@@ -27,7 +27,27 @@ use libc::{c_char, c_double, c_float, c_int, c_short, c_uchar, c_void, intmax_t,
 // and alias it ourselves.
 pub type pid_t = libc::c_int;
 
-pub type Lisp_Object = EmacsInt;
+#[repr(C)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct Lisp_Object(EmacsInt);
+
```
Comment (Wilfred) [inline rust_src/remacs-sys/lib.rs]: Have you tested this on 32-bit? This has caused problems in the past: https://github.com/rust-lang/rfcs/pull/1758#issuecomment-276465077

Comment (shaleh) [reply rust_src/remacs-sys/lib.rs]: Sigh. No, I didn't.

That,RFC merged. In theory I can switch this to transparent now.

Comment (Wilfred) [reply rust_src/remacs-sys/lib.rs]: I've created #568 to keep track of this.

**Pull request merged**